ext/ldap/ldap.c

```diff
@@ -3189,6 +3210,17 @@ PHP_FUNCTION(ldap_set_option)
 #endif
 #ifdef LDAP_OPT_X_TLS_PROTOCOL_MIN
 	case LDAP_OPT_X_TLS_PROTOCOL_MIN:
+#endif
+		LDAPG(newctx) = true;
```

Comment:
I think those options make sense only if link is NULL. So maybe adding `if (link == NULL)` would make sense as there is not much point to do reset if someone tries to do it on the actual connection. I guess in such case we should not probably even attempt to do any setting as it is pointless but that's more optimization I guess.

Reply:
`ldap_set_option` may be called for a link, before `ldap_start_tls`